azure-templates/steps-pre-build.yml

```diff
@@ -115,8 +117,14 @@ steps:
     displayName: Check out NI VS Custom Device Build Tools repository
 
     # The checkout order doesn't matter in most cases, but we want the dev branch (self) to be checked out last so that it's used for PR builds in niveristand-custom-device-build-tools
-  - checkout: self
-    displayName: Check out repository to build
+  - ${{ if eq(parameters.submodules, true) }}:
+    - checkout: self
+      displayName: Check out repository to build with submodules
+      submodules: true
```

Comment:
Instead of the two if statements, could you just use the variable value itself for `submodules`?

Reply:
yes.  Originally that parameter was a string rather than a boolean so it didn't work, but now that it's a boolean the if logic is unneeded, good catch.

Reply:
95f926dc1c3cb9d3a4fa96c829d5d1cb649a91f5